proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+  Phabricator documentation for this https://secure.phabricator.com/book/phabricator/article/configuring_accounts_and_registration/
+  
+- Link Trac to GitHub
+
+  We already have a read-only mirror for code that is incredibly useful. I propose extending this read-only mode further. 
+
+   * Support OAuth on Trac for logins against GitHub as well.
+   * Synchronize Trac issues in a read-only way to GitHub. The usefulness of this feature comes In the ability for users to reference the issues from their own GitHub projects. This gives us some valuable information such as who these issues affect. This information would get synched back to Trac. This also allows users to be notified when something gets fixed. Since the issue would be closed on GitHub as well. Finally it gives some information on the importance of issues.
+
+  With the ability to login directly on Trac using your GitHub account, I believe it's easier to take issues on one place: Trac. So read-only would be enough.
+
+  As a whole, I find Trac MUCH better for ticket triaging than Github or Phab,
+  both of which seem to be quite bare and simple in what they provide. I am not
+  in favor of ditching it. Whatever we gain we lose in the ability to generate custom views on issues. Which is a powerful filtering mechanism for prioritization.
+
+  Also we have and continue to accept patches just uploaded
+  to Trac as a diff. We tend to ask people to upload it to phab for better reviews
+  and so it's attributed to them when we commit. Some don't (and we then do it ourselves),
+  most due. If they don't need another login then I suspect almost all would.
+
+  There's a (seemingly) actively maintained project that does all the above, could we leverage it?
+  https://github.com/trac-hacks/trac-github
+
+- Allow doc changes on Trac
+
+  I am not sure about the feasibility of this one, but there is a Trac plugin that adds a new section to trac under /doc that is able to render .rst files and more from a source repository. Essentially making a wiki from them.
+
+  https://github.com/trac-hacks/tracdocs
+
+  We could have harbormaster build the docs and provide them to trac. I am not sure how reviews would go. Styling would be another issue. 
+
+  Lastly this plugin seems to only support SVN. but that's not a huge barrier.
+
+- Helping Newcomers
+
+   * Create a better landing page for newcomers (see proposal below)
+   * Show newcomers tickets more prominently
+   * Clean up build instructions. The pages have become way too complex over the years. 
+   * Create native setup script particular to an environment.
+
+  As I am a windows maintainer, I will offer a proposal: Use scripts similar to my own setup scripts. my setup script for a 100% unattended build env setup for Windows are here: https://github.com/Mistuke/GhcDevelChoco/releases
```

Comment:
It could be for anyone who wants to set up a GHC dev environment on Windows. From what I've noticed the past year, a lot of the difficulties of setting of setting up the environment come from people setting things up from either a "dirty" msys2 environment (e.g. one that's already had a bunch of packages and changes done to it) or using a stage0 compiler installed in a place with spaces etc. 

This script gives you a working environment which you can customize after that. The SSH feature was added so that Linux programmers who want to test things on Windows don't need to know Windows. Just run the one powershell command and ssh into the VM/Box.